### PR TITLE
[skip ci] Upgrade GHC to 8.8.4

### PR DESCRIPTION
**Pull request by akshaymankar**
Fixes https://github.com/zinfra/backend-issues/issues/1622

### CODE REVIEW

**Comment by akshaymankar** [comment]
Special pipeline here: https://bridge-ie-concourse.zinfra.io/teams/main/pipelines/wire-server-ghc88.

File: stack.yaml
Comment (jschaul) [inline stack.yaml:58]: can this be removed? Or why is it commented?

File: stack.yaml
Comment (jschaul) [inline stack.yaml:67]: The various forks should have a reference to a stable branch, not a PR branch which will be deleted after merging the PR. Perhaps before merging this PR; merge the upstream ones and use a stable branch commit identifier?

**Review by jschaul — APPROVED** [review]
Generally looks good.

Two things:

1. can the dependencies be based on merged commits, rather than hanging branches that may disappear by accident?

> Special pipeline here: https://bridge-ie-concourse.zinfra.io/teams/main/pipelines/wire-server-ghc88.

2. can the ubuntu-based Docker image also be tested with ghc-8.8? The pipeline seems to include an ubuntu image as well; however I cannot find the pipeline code that was used to create this pipeline anywhere. Please link.

File: stack.yaml
Comment (akshaymankar) [reply stack.yaml:67]: This one is the only one we control. I created PR here: https://github.com/wireapp/saml2-web-sso/pull/66
No other branches will get deleted on merge, I have set the option to not delete it in Gitlab. For github forks we don't need to worry as github only deletes branches in the same repo. 

I suspect some of the upstream maintainers aren't maintaining the packages very actively, so it might take very long. So, I think it is fine to refer to these packages in their forks.

File: stack.yaml
Comment (akshaymankar) [reply stack.yaml:58]: Hmm, I forgot a few comments, I will clean them up, thanks!

**Pull request merged**
